lua/telescope/utils.lua

```diff
@@ -147,6 +147,37 @@ utils.path_tail = (function()
   end
 end)()
 
+utils.is_path_hidden = function(opts)
+  local path_display = utils.get_default(opts.path_display, require('telescope.config').values.path_display)
+
+  return path_display == nil or path_display == "hidden" or
+    type(path_display) ~= "table" or vim.tbl_contains(path_display, "hidden")
+end
+
+utils.transform_path = function(opts, path)
+  if utils.is_path_hidden(opts) then
+    return ''
+  end
+
+  local path_display = utils.get_default(opts.path_display, require('telescope.config').values.path_display)
```

Comment:
It feels weird that we are doing `utils.get_default` twice. Once in is_path_hidden and again immediately again

Reply:
Yes, feels a little awkward, but I think it makes sense given that `is_path_hidden` should be separated into a general util function. 

I could do 
```
  return (path_display == nil or path_display == "hidden" or
    type(path_display) ~= "table" or vim.tbl_contains(path_display, "hidden")), path_display
```

and transform_path could do 
```
local is_path_hidden, path_display = utils.is_path_hidden(opts)
```

and other functions could just use the first value.

I'm personally iffy on this approach, since returning the extra value will not align with the name of the function, but open to other opinions/suggestions.

Reply:
Maybe you could make `path_display` an optional parameter for `is_path_hidden`?
Then if it is passed to `is_path_hidden`, we use it, if not, we calculate it in the way you have done.

This means we can calculate it once in `transform_path`, before using `is_path_hidden`, then pass it to `is_path_hidden` where it is not calculated for a second time.

This allows the current functionality of `is_path_hidden` to be retained when no `path_display` parameter is passed, but avoids having to calculate the same thing multiple times in quick succession.

_(Hope this comment makes sense, it feels a little rambly, but not sure how else to phrase it 😅)_

Reply:
Thanks @l-kershaw , I've added this to the most recent commit. I'm not 100% sold on it, since it feels a little awkward that after passing `path_display`, `opt` argument is no longer used. But feels slightly better than the current version.

Reply:
@caojoshua I agree, it does feel a little awkward.
Maybe we could make it clearer for future development by adding documentation to `is_path_hidden` to explain the intention.

(and/or)

Perhaps we could add a check that errors if both `opt` and `path_display` are non-`nil`, as it is ambiguous which parameter should take precedence in that situation?
Although this is making what should be quite a simple function even more complicated, so I don't really know what's best here.